x-pack/legacy/plugins/epm/public/screens/detail/header.tsx

```diff
@@ -3,52 +3,76 @@
+import { EuiFlexGroup, EuiFlexItem, EuiPage, EuiTitle, IconType } from '@elastic/eui';
 import { PLUGIN } from '../../../common/constants';
 import { PackageInfo } from '../../../common/types';
 import { Version } from '../../components/version';
 import { IconPanel } from '../../components/icon_panel';
-import { useBreadcrumbs, useCore, useLinks } from '../../hooks';
+import { useBreadcrumbs, useLinks } from '../../hooks';
 import { CenterColumn, LeftColumn, RightColumn } from './layout';
+import { InstallationButton } from '../../components/installation_button';
+import { ConfirmPackageInstall } from '../../components/confirm_package_install';
+// TODO: either
+// * import from a shared point
+// * duplicate inside our folder
+import { useTrackedPromise } from '../../../../infra/public/utils/use_tracked_promise';
+import { installPackage } from '../../data';
+import { NavButtonBack } from '../../components/nav_button_back';
+
+const FullWidthNavRow = styled(EuiPage)`
+  /* no left padding so link is against column left edge  */
+  padding-left: 0;
+`;
+
+const Text = styled.span`
+  margin-right: ${props => props.theme.eui.euiSizeM};
+`;
+
+const StyledVersion = styled(Version)`
+  font-size: ${props => props.theme.eui.euiFontSizeS};
+  color: ${props => props.theme.eui.euiColorDarkShade};
+`;
 
 type HeaderProps = PackageInfo & { iconType?: IconType };
 
 export function Header(props: HeaderProps) {
-  const { iconType, title, version } = props;
-  const { theme } = useCore();
+  const [isModalVisible, setModalVisible] = useState<boolean>(false);
+  const { iconType, title, version, assets } = props;
   const { toListView } = useLinks();
   useBreadcrumbs([{ text: PLUGIN.TITLE, href: toListView() }, { text: title }]);
+  const [isInstalled, setInstalled] = useState<boolean>(props.status === 'installed');
+  const [installationRequest, attemptInstallation] = useTrackedPromise(
+    {
+      createPromise: async () => {
+        const pkgkey = `${props.name}-${props.version}`;
+        return installPackage(pkgkey);
+      },
+      onResolve: (value: PackageInfo) => setInstalled(value.status === 'installed'),
+    },
+    [props.name, props.version, installPackage]
+  );
+  const isLoading = installationRequest.state === 'pending';
 
-  const FullWidthNavRow = styled(EuiPage)`
-    /* no left padding so link is against column left edge  */
-    padding-left: 0;
-  `;
+  const handleModal = useCallback(() => {
+    setModalVisible(!isModalVisible);
+  }, [isModalVisible]);
 
-  const Text = styled.span`
-    margin-right: ${theme.eui.euiSizeM};
-  `;
+  const handleClickInstall = useCallback(() => {
+    attemptInstallation();
+    handleModal();
+  }, [attemptInstallation, handleModal]);
 
-  const StyledVersion = styled(Version)`
-    font-size: ${theme.eui.euiFontSizeS};
-    color: ${theme.eui.euiColorDarkShade};
-  `;
+  const numOfAssets = useMemo(
+    () =>
+      Object.entries(assets).reduce(
```

Comment:
I'd love to have this logic exist in a single place outside the view so we don't duplicate logic, but can't think of an appropriate place right now. 

In the meantime, can you filter keys to only count `KibanaAssetType` values since that's all we're currently installing? In the `coredns` example from the screenshot, that will exclude `infrastructure-ui-source` and give a count of 14 instead of 15. We can verify that by checking `savedObject.attributes.installed.length` from the http://localhost:5601/api/epm/package/coredns-1.0.1 response

I'd also be happy to add an `InstallableAssetTypes` type or something similar. Let me know if you have any other ideas.

Reply:
If we don't need certain assets during installation and we're not going to show them to the user, can we filter them out before it gets sent to the client?  Then we wouldn't need to bother with filtering and can have a separate type you suggest.

Reply:
Oh, yeah. That's what the API response is for. Thanks. Let's consolidate the logic for what's installable in `groupPathsByService`. That works for now and likely works as we add types. If not we can always revisit, but at least we're starting with the logic in one place. Nice catch.

Reply:
That sounds good, though I think we should put this in a separate PR as I don't think I need to make changes to this modal, the code should work as is, but  I will need to change the assets section.